discord/ext/commands/core.py

```diff
@@ -70,6 +98,22 @@
     'bot_has_guild_permissions'
 )
 
+MISSING: Any = discord.utils.MISSING
+
+T = TypeVar('T')
+CT = TypeVar('CT', bound='Cog')
+CMT = TypeVar('CMT', bound='Command')
+CXT = TypeVar('CXT', bound='Context')
+# CHT = TypeVar('CHT', bound='Check')
+GT = TypeVar('GT', bound='Group')
+HT = TypeVar('HT', bound='Hook')
+ET = TypeVar('ET', bound='Error')
```

Comment:
Again with the, typevar naming convention.

I think abbreviated PascalCase With a T on the end e.g. `CtxT`, `CogT`, `GrpT` is probably the way to go?